docs/core/diagnostics/app_is_leaking_memory_eventual_crash.md

```diff
@@ -0,0 +1,131 @@
+      System.Runtime:
+        CPU Usage (%)                                  4
+        Working Set (MB)                              66
+        GC Heap Size (MB)                              4
+        Gen 0 GC / second                              0
+        Gen 1 GC / second                              0
+        Gen 2 GC / second                              0
+        Number of Exceptions / sec                     0
+```
+
+Here we can see that right after startup, the managed heap memory is 4 MB.
+
+Now, let's hit the URL [http://localhost:5000/api/diagscenario/memleak/200000](http://localhost:5000/api/diagscenario/memleak/200000)
+
+Rerun the dotnet-counters command. We should see an increase in memory usage as shown below:
+
+```console
+    Press p to pause, r to resume, q to quit.
+      System.Runtime:
+        CPU Usage (%)                                  4
+        Working Set (MB)                             353
+        GC Heap Size (MB)                            258
+        Gen 0 GC / second                              0
+        Gen 1 GC / second                              0
+        Gen 2 GC / second                              0
+        Number of Exceptions / sec                     0
+```
+
+Memory has now grown to 258 MB.
+
+We can safely say that memory is growing or leaking. The next step is collect the right data for memory analysis.
+
+### Core dump generation
+
+When analyzing possible memory leaks, we need access to the apps memory heap. We then analyze the memory contents. Looking at relationships between objects, we create theories on why memory isn't being freed. A common diagnostics data source is a memory dump (Win) or the equivalent core dump (Linux). To generate a core dump of a .NET Core application, we can use the [dotnet-dump)](dotnet-dump.md) tool.
+
+Using the previous [Sample debug target](sample-debug-target.md) started above, run the following command to generate a core dump:
+
+```dotnetcli
+sudo dotnet-dump collect -p 4807
+```
+
+4807 is the process ID that can be found using `dotnet-trace list-processes`. The result is a core dump located in the same folder.
+
+> [!NOTE]
+> To generate core dumps, `dotnet-dump` requires sudo.
+
+### Analyzing the core dump
+
+Now that we have a core dump generated, use the [dotnet-dump)](dotnet-dump.md) tool to analyze the dump:
+
+```dotnetcli
+dotnet-dump analyze core_20190430_185145
+```
+
+Where `core_20190430_185145` is the name of the core dump you want to analyze.
+
+> [!NOTE]
+> If you see an error complaining that libdl.so cannot be found, you may have to install the libc6-dev package.
+
+You'll be presented with a prompt where you can enter SOS commands. Commonly, the first thing we want to look at is the overall state of the managed heap:
+
+```console
+> dumpheap -stat
+```
+
+The (partial) output can be seen below:
+
+![alt text](https://user-images.githubusercontent.com/15442480/57110756-7d32ac80-6cee-11e9-9b80-2ce700e7a2f1.png)
```

Comment:
You should add alt text for accessibility reasons and these images should be added to the repo under 
docs/core/diagnostics/media/\<name-article>/friendly-image-name-with-hyphens-for-seo.png

This applies to all images here.